src/inmanta/data/__init__.py

```diff
@@ -2546,9 +2546,27 @@ def set_connection_pool(pool):
 
 
 async def connect(
-    host: str, port: int, database: str, username: str, password: str, create_db_schema: bool = True
+    host: str,
+    port: int,
+    database: str,
+    username: str,
+    password: str,
+    create_db_schema: bool = True,
+    connection_pool_min_size: int = 10,
+    connection_pool_max_size: int = 10,
+    database_connection_timeout=60,
 ) -> asyncpg.pool.Pool:
-    pool = await asyncpg.create_pool(host=host, port=port, database=database, user=username, password=password)
+    connection_kwargs = {"timeout": database_connection_timeout}
```

Comment:
Putting it as a keyword argument in the method call would improve readability. 

We should also add a test that set the config options on the Inmanta server.